### PR TITLE
Minor cleanup of dartsim

### PR DESCRIPTION
**Pull request by shameekganguly**
- Change remaning ignerr to gzerr
- Clangtidy fixes
- Remove unused header

### CODE REVIEW

File: dartsim/src/ShapeFeatures.hh [hunk 1]
```diff
@@ -148,7 +148,7 @@ class ShapeFeatures :
   public: Identity AttachEllipsoidShape(
       const Identity &_linkID,
       const std::string &_name,
-      const Vector3d _radii,
+      Vector3d _radii,
```
Comment (iche033) [inline dartsim/src/ShapeFeatures.hh:151]: I think it was meant to be `const Vector3d &_radii`

Comment (shameekganguly) [reply dartsim/src/ShapeFeatures.hh]: Done

File: dartsim/src/ShapeFeatures.cc [hunk 2]
```diff
@@ -252,7 +252,7 @@ Vector3d ShapeFeatures::GetEllipsoidShapeRadii(
 Identity ShapeFeatures::AttachEllipsoidShape(
     const Identity &_linkID,
     const std::string &_name,
-    const Vector3d _radii,
+    const Vector3d &_radii,
```
Comment (azeey) [inline dartsim/src/ShapeFeatures.cc:255]: I think this is a breaking change, so we wouldn't want to do it in this branch. Instead, it should go into `main`. Also, to do this, we'd have to change the signatures in the `AttachEllipsoidShapeFeature` (https://github.com/gazebosim/gz-physics/blob/8ee8028861567303cdba2602c9ba79c34a5ab542/include/gz/physics/EllipsoidShape.hh)

Comment (scpeters) [reply dartsim/src/ShapeFeatures.cc:255]: I've applied these API changes in https://github.com/gazebosim/gz-physics/pull/521, targeting `main`

Comment (scpeters) [reply dartsim/src/ShapeFeatures.cc]: I reverted this change in https://github.com/gazebosim/gz-physics/pull/515/commits/39e475315769ed20a77e5c2e8ff1e3fb32795f30

**Review by iche033 — APPROVED** [review]

**Comment by shameekganguly** [comment]
Thanks Steve, looks good!

**Pull request merged**